lib/WebSocketServer.js

```diff
@@ -40,7 +40,7 @@ function WebSocketServer(options, callback) {
     maxPayload: null
   }).merge(options);
 
-  if (!options.isDefinedAndNonNull('port') && !options.isDefinedAndNonNull('server') && !options.value.noServer) {
+  if (!options.value.noServer && (!options.isDefinedAndNonNull('port') || !options.isDefinedAndNonNull('server'))) {
```

Comment:
This means we should not make the Server which is listing on a port as the server option?